### DISCUSSION TITLE
q-range /q-slider with exponential step

### DISCUSSION THREAD

**Discussion by gaetan-cordonnier** [quasarframework/quasar#15007] [Ideas / Proposals] [answered]
Hello, I have been using Quasar for some time and I have a problem with the 'q-range'. When the numbers are very large and you want to select small values but also very large values, it's complicated.
example:
https://codepen.io/master999/pen/PoadEQg

What is needed is to be able to have several different 'step' ex: for a min 0 and max 1,000,000:
0 to 10 with step 1, 10 to 100 with step 10, 100 to 1000 with step 100...
demo:
https://stackblitz.com/edit/vue-simple-range-slider-example?embed=1&file=src/App.vue&view=preview
(github: https://github.com/mohamadkhani/vue-simple-range-slider)

This could be an interesting upgrade to the quasar q-range component.

I am currently using the vue-simple-range-slider library as a replacement for quasar q-range but it doesn't work on mobile.

**Comment by gaetan-cordonnier** [comment]
Other sample code: https://codepen.io/janwirth/pen/ZJEKxp

**Comment by pdanpdan** [comment, marked as answer]
You can use manipulation of the model

https://codepen.io/pdanpdan/pen/XWYPZMq?editors=0110

you have there a composable for using an exponential model with QRange

**Reply by g-cordonnier** [reply]
Great! Thank you @pdanpdan it's work ! Good job
